drivers/serial/uart_nrfx_uart.c

```diff
@@ -309,6 +370,8 @@ static void uart_nrfx_isr(void *arg)
 	if (m_irq_callback) {
 		m_irq_callback(dev);
 	}
+
+	device_busy_clear(dev);
```

Comment:
This cannot be called here, since `uart_nrfx_fifo_fill` is supposed to be called from the callback. In the current form, right after some byte is passed to the transmitter (by `uart_nrfx_fifo_fill`), the device is marked as not busy, although the transmission of the byte may still be in progress. Instead, the busy status should be cleared when the next IRQ occurs and nothing more is passed to the transmitter (i.e. `uart_nrfx_fifo_fill` is not called again). Thus, the clearing should be done in `uart_nrfx_irq_tx_disable`. Consequently, it would be better to call `device_busy_set` from `uart_nrfx_irq_tx_enable`, not from `uart_nrfx_fifo_fill`.